src/include/duckdb.h

```diff
@@ -2339,6 +2339,14 @@ Fetch the internal arrow schema from the prepared statement.
 */
 DUCKDB_API duckdb_state duckdb_prepared_arrow_schema(duckdb_prepared_statement prepared,
                                                      duckdb_arrow_schema *out_schema);
+/*!
+Convert a data chunk into an arrow array.
+
+* result: The result object the data chunk have been fetched from.
+* chunk: The data chunk to convert.
+* out_array: The output array.
+*/
+DUCKDB_API void duckdb_result_arrow_array(duckdb_result result, duckdb_data_chunk chunk, duckdb_arrow_array *out_array);
```

Comment:
Shouldn't this need an index?
DataChunks hold multiple Vectors

Unless you're creating a struct containing the columns of the chunk, in that case you would still need names for the struct fields ? I guess that's taken from the result then

Reply:
I'm returning the same arrow array as in the current C-API, a struct array that doesn't need an index.

We could add a function to export only one column from the datachunk

Reply:
Yea I saw that later, I'm not a huge fan of that ambiguity, it's unclear what it does judging from the inputs alone
But that's not really a comment on your PR, I think it's fine to follow the existing design 👍